community_usecase/Profile_generation/run_profile_generation.py

```diff
@@ -0,0 +1,394 @@
+    biography: str = Field(
+        description="Narrative biography including career trajectory and "
+                    "research journey")
+
+
+class ResearchInterests(BaseModel):
+    areas: list[str] = Field(
+        description="List of active or long-term research topics")
+
+
+class AwardsAndDistinctions(BaseModel):
+    honors: list[str] = Field(
+        description="List of awards, honors, and professional distinctions "
+                    "received")
+
+
+class Education(BaseModel):
+    degrees: list[str] = Field(
+        description="Academic degrees in reverse chronological order, "
+                    "including institution and year")
+
+
+class Links(BaseModel):
+    scholarly_identity_links: list[str] = Field(
+        description="Links to unique scholarly identity profiles (e.g., "
+                    "ORCID, IEEE Xplore, DBLP)"
+    )
+    related_sites: list[str] = Field(
+        description="Links to institutional or departmental homepages (e.g., "
+                    "KAUST ECE department)"
+    )
+    related_links: list[str] = Field(
+        description="Links to academic and professional presence (e.g., "
+                    "Google Scholar, ResearchGate, LinkedIn)"
+    )
+
+
+class ScholarProfile(BaseModel):
+    personal_information: PersonalInformation
+    biography: Biography
+    research_interests: ResearchInterests
+    awards_and_distinctions: AwardsAndDistinctions
+    education: Education
+    links: Links
+
+
+def generate_html_profile(input_text,
+                          template_path: str = "template.html",
+                          output_file: str = "profile.html",
+                          base_url: str = "https://cemse.kaust.edu.sa") -> \
+        None:
+    """Fill an HTML template with profile data, some with agent rewriting,
+    and write to file."""
+    extraction_prompt = (
+        "You are a scholarly‑profile extraction assistant. "
+        "Return ONLY JSON that matches the provided schema."
+    )
+
+    extractor = ChatAgent(system_message=extraction_prompt)
+    extraction_response = extractor.step(
+        input_message=input_text,
+        response_format=ScholarProfile,
+    )
+    profile = extraction_response.msgs[0].parsed
+
+    def to_html_list(items: list[str]) -> str:
+        return "<br>".join(items)
+
+    def format_awards_with_agent(awards: list[str]) -> str:
+        formatted_awards = []
+        agent = ChatAgent(
+            system_message="""Reformat awards as HTML list items using the 
+                           structure, Like:
+                           <li class="field__item"><strong>The 
+                           NSF CAREER grant in low-power computing and 
+                           communication systems</strong>, 2024</li>.""")
+        for award in awards:
+            response = agent.step(f"Reformat this: {award}")
+            formatted_awards.append(response.msgs[0].content.strip())
+        return "\n".join(formatted_awards)
```

Comment:
Do not do it in multiple steps. Use response format to format the awards in one shot.

Reply:
The format problem can be solve by adding examples in the `Field` `description`.